dao/src/main/java/org/thingsboard/server/dao/sql/alarm/AlarmRepository.java

```diff
@@ -345,4 +347,13 @@ String updateAlarm(@Param("t_id") UUID tenantId, @Param("a_id") UUID alarmId, @P
 
     @Query(value = "SELECT unassign_alarm(:t_id, :a_id, :a_ts)", nativeQuery = true)
     String unassignAlarm(@Param("t_id") UUID tenantId, @Param("a_id") UUID alarmId, @Param("a_ts") long unassignTime);
+
+    @Query(value = "SELECT at.type FROM alarm_types AS at WHERE at.tenant_id = :tenantId AND LOWER(at.type) LIKE LOWER(CONCAT('%', :searchText, '%'))", nativeQuery = true)
```

Comment:
In Postgres you should use `ILIKE` instead of the expensive full-scan computation `LOWER(at.type) LIKE LOWER(...)`
The second reason is `ilike` can hit GIN or GIST indexes. That probably you can use as well.

Reply:
Let's leave ILIKE and remove [pg_trgm](http://www.postgresql.org/docs/11/static/pgtrgm.html) extension from the upgrade script. I am unsure how pg_trgm works in the context of the unique constraint. The index declaration does not have anything about the tenant_id column.